### PR TITLE
docs: Sync readme with go-pdk

### LINKED ISSUE #11: Cannot find main entrypoint CS5001
```
CSC : error CS5001: Program does not contain a static 'Main' method suitable for an entry point [/Users/ben/dylibso/hello-from/hello-from-csharp/HelloFromCsharp.csproj]
```

Tried to compile a pure library and kept getting this error. I thought setting `<OutputType/>` to `Library` might resolve but it did not. What i ended up doing as adding a fake instruction on the top of the file to get the compiler to generate a main, which i then don't use: https://github.com/extism/hello-from/blob/main/hello-from-csharp/Program.cs#L5

### PR DESCRIPTION
**Pull request by mhmd-azeez**
Todo: test host functions example

Fixes #11

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -39,214 +39,418 @@ Update your MyPlugin.csproj as follows:
...
+        count <- BitConverter.ToInt32(buffer.ReadBytes())
+    
+    count <- count + 1
+    let countBytes = BitConverter.GetBytes(count)
+    
+    Pdk.SetVar("count", countBytes)
+    Pdk.SetOutput(count.ToString())
+    
+    0
+```
```
Comment (NickDarvey) [inline README.md:302]:
```suggestion
```fsharp
[<UnmanagedCallersOnly>]
let count () =

    let count =
        match Pdk.TryGetVar "count" with
        | true, buffer ->
            BitConverter.ToInt32(buffer.ReadBytes())
        | false, _ ->
            0
    
    let count = count + 1

    Pdk.SetVar("count", BitConverter.GetBytes(count))
    Pdk.SetOutput(count.ToString())
    
    0
```
```

**Review by NickDarvey — APPROVED** [review]
Only minor tweaks, and I'm not sure they're even necessary. Honestly, your F# samples are pretty much what I'd write.

File: README.md [hunk 2]
```diff
@@ -39,214 +39,418 @@ Update your MyPlugin.csproj as follows:
...
 
-int.TryParse(Encoding.UTF8.GetString(totalBlock.ReadBytes()), out var total);
+F#:
+```fsharp
+[<UnmanagedCallersOnly(EntryPoint = "greet")>]
+let Greet () =
+    let name = Pdk.GetInputString()
+    if name = "Benjamin" then
+        Pdk.SetError("Sorry, we don't greet Benjamins!")
```
[the remark this answers is not in the document]
Comment (bhelx) [reply README.md:141]: this is an outrage!

File: README.md [hunk 3]
```diff
@@ -39,214 +39,418 @@ Update your MyPlugin.csproj as follows:
...
+[<UnmanagedCallersOnly(EntryPoint = "greet")>]
+let Greet () =
+    let mutable user : string = null
+    let success = Pdk.TryGetConfig( "user", &user)
+    if not success then
+            raise (new System.Exception(sprintf "Key '%s' not found" "user"))
+
+    let greeting = $"Hello, {user}!"
+    Pdk.SetOutput(greeting)
     0
```
Comment (bhelx) [inline README.md:256]:
```suggestion
[<UnmanagedCallersOnly(EntryPoint = "greet")>]
let Greet () =
    match Pdk.TryGetConfig "user" with
    | true, user ->
        let greeting = $"Hello, {user}!"
        Pdk.SetOutput(greeting)
        0
    | false, _ ->
        failwith "This plug-in requires a 'user' key in the config"
```

File: README.md [hunk 4]
```diff
@@ -39,214 +39,418 @@ Update your MyPlugin.csproj as follows:
...
 ```
```
[the remark this answers is not in the document]
Comment (bhelx) [reply README.md]: Committed these suggestions, thank you!

File: README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (mhmd-azeez) [reply README.md:141]: hahaha i guess it's not their lucky day

**Comment by mhmd-azeez** [comment]
@NickDarvey thank you very much for your review

**Comment by mhmd-azeez** [comment]
@bhelx If there is no more feedback, feel free to merge this PR

Notes:
1) It's okay for the tests to fail, the new Extism CLI now uses the `extism:host/env` namespace. This is fixed in #16 
2) I changed the host to Go because I had a problem with the python SDK:
```
Traceback (most recent call last):
  File "D:\x\python\host\main.py", line 22, in <module>
    Function(
  File "C:\Users\muham\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\extism\extism.py", line 212, in __init__
    self.pointer = _lib.extism_function_new(
                   ^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: initializer for ctype 'void(*)(struct ExtismCurrentPlugin *, ExtismVal *, unsigned long long, ExtismVal *, unsigned long long, void *)' must be a cdata pointer, not function
```

**Review by bhelx — APPROVED** [review]

**Pull request merged**